lib/apiservers/engine/backends/cache/image_cache.go

```diff
@@ -209,3 +174,43 @@ func (c *ImageCache) GetImageByNamed(named reference.Named) (*metadata.ImageConf
 
 	return config, nil
 }
+
+// AddImage adds an image to the image cache
+func (c *ImageCache) AddImage(imageConfig *metadata.ImageConfig) {
+	c.m.Lock()
+	defer c.m.Unlock()
+	var imageID string
+
+	// Don't assume the image id in image has "sha256:<id> as format.  We store it in
+	// this fomat to make it easier to lookup by digest
+	if strings.HasPrefix(imageConfig.ImageID, "sha") {
```

Comment:
Could the prefix be something other than sha256? (why not check for `sha256` as prefix?)